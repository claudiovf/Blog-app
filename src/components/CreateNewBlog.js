import React, { useState } from 'react'
import { addNewBlog } from '../reducers/blogReducer'
import { connect } from 'react-redux'
import { Button } from 'react-bootstrap'

const CreateNewBlog = (props) => {
	const [newTitle, setNewTitle] = useState('')
	const [newAuthor, setNewAuthor] = useState('')
	const [newUrl, setNewUrl] = useState('')

	const handleTitleChange = event => {
		setNewTitle(event.target.value)
	}
	const handleAuthorChange = event => {
		setNewAuthor(event.target.value)
	}
	const handleUrlChange = event => {
		setNewUrl(event.target.value)
	}

	const addNew = event => {
		event.preventDefault()

		props.addNewBlog({
			title: newTitle,
			author: newAuthor,
			url: newUrl,
			likes: 0
		})

		setNewTitle('')
		setNewAuthor('')
		setNewUrl('')

	}


	return (
		<form onSubmit={addNew} className="blogForm">
			<h2>Create New Blog</h2>
			<div>
            Title:
				<input
					type="text"
					value={newTitle}
					name="New Title"
					onChange={input => handleTitleChange(input)}
				/>
			</div>
			<div>
            Author:
				<input
					type="text"
					value={newAuthor}
					name="New Author"
					onChange={input => handleAuthorChange(input)}
				/>
			</div>
			<div>
            Url:
				<input
					type="text"
					value={newUrl}
					name="New Url"
					onChange={input => handleUrlChange(input)}
				/>
			</div>
			<Button variant="dark" name="Blog Submit" type="submit">Create</Button>
		</form>
	)
}

const ConnectedCreateNewBlog = connect(
	null,
	{addNewBlog}
)(CreateNewBlog)
export default ConnectedCreateNewBlog