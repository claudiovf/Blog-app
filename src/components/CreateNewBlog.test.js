import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import CreateNewBlog from './CreateNewBlog'

describe('<CreateNewBlog /> ', () => {
	let component
	const createBlog = jest.fn()

	const tree = renderer
		.create(<CreateNewBlog createBlog={createBlog} />)
		.toJSON()

	expect(tree).toMatchSnapshot()

	beforeEach(() => {

		component = render(
			<CreateNewBlog createBlog={createBlog} />
		)
	})

	test('receives the right info from submit', () => {

		const form = component.container.querySelector('.blogForm')
		const title = component.container.querySelector('[name="New Title"]')
		const author = component.container.querySelector('[name="New Author"]')
		const url = component.container.querySelector('[name="New Url"]')


		fireEvent.change(title, {
			target: { value: 'Testing BlogForm with Jest' }
		})
		fireEvent.change(author, {
			target: { value: 'Jest' }
		})
		fireEvent.change(url, {
			target: { value: 'www.bbc.com' }
		})

		fireEvent.submit(form)

		expect(createBlog.mock.calls).toHaveLength(1)
		expect(createBlog.mock.calls[0][0].title).toBe('Testing BlogForm with Jest')
		expect(createBlog.mock.calls[0][0].author).toContain('Jest')
		expect(createBlog.mock.calls[0][0].url).toContain('www.bbc.com')

	})

})