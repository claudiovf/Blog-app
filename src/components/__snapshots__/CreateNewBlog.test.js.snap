// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<form
  className="blogForm"
  onSubmit={[Function]}
>
  <h2>
    Create New Blog
  </h2>
  <div>
    Title:
    <input
      name="New Title"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    Author:
    <input
      name="New Author"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    Url:
    <input
      name="New Url"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <button
    type="submit"
  >
    Create
  </button>
</form>
`;
